lib/sqeduler/lock_maintainer.rb

```diff
@@ -0,0 +1,82 @@
+module Sqeduler
+  # This is to ensure that if you set your jobs to run one a time and something goes wrong
+  # causing a job to run for a long time, your lock won't expire.
+  # This does't stop long running jobs, it just ensures you only end up with one long running job
+  # rather than 20 of them.
+  class LockMaintainer
+    RUN_INTERVAL = 30
+
+    def initialize
+      @class_with_locks = {}
+    end
+
+    # This is only done when we initialize Sqeduler, don't need to worry about threading
+    def run
+      @maintainer_thread ||= Thread.new do
+        loop do
+          begin
+            synchronize
+          rescue => ex
+            Service.logger.error "[MAINTAINER] Error while trying to refres locks, #{ex.class}, #{ex.message}"
```

Comment:
Will this get prefixed with anything else? `[MAINTAINER]` will be quite a mystery by itself :)

`[SQEDULER LOCK MAINTAINER]` maybe?